### ISSUE TITLE
ResourceKnownHostsServerKeyVerifier does not consider the specified keytypes

### ISSUE THREAD

**Issue by ruedih** [spring-projects/spring-integration#8674]
**In what version(s) of Spring Integration are you seeing this issue?**
6.0.5.
6.1.1 should still be affected as the code was not changed

**Describe the bug**
The ResourceKnownHostsServerKeyVerifier in Spring Integration Sftp seems to always select the first matching key from the provided known_host file and seems not to take into account the specified keytype.

**To Reproduce**

To reproduce the issue, add multiple keytypes for the same host and port. Then experiment with the order.

**Expected behavior**

The keytype should be considered, and the [findKnownHostEntry method](
https://github.com/spring-projects/spring-integration/blob/v6.1.1/spring-integration-sftp/src/main/java/org/springframework/integration/sftp/session/ResourceKnownHostsServerKeyVerifier.java#L109) should select the matching host, port and keytype.

**Related Code **
You can refer to the code at https://github.com/spring-projects/spring-integration/blob/v6.1.1/spring-integration-sftp/src/main/java/org/springframework/integration/sftp/session/ResourceKnownHostsServerKeyVerifier.java#L122

**Comment by artembilan** [comment]
Yeah... We blindly modeled this one after `KnownHostsServerKeyVerifier` in our currently used version of `sshd-core`.
Looks like we have just missed their TODO over there in the code:
```
// TODO allow for several candidates and check if ANY of them matches the key and has 'revoked' marker
HostEntryPair match = findKnownHostEntry(clientSession, remoteAddress, knownHosts);
```
Now in the latest version I see they have it fixed:
```
https://github.com/apache/mina-sshd/blob/master/sshd-core/src/main/java/org/apache/sshd/client/keyverifier/KnownHostsServerKeyVerifier.java#L273C8-L273C8
```

Will fix ours shortly respectively.

Thank you for the pointer!

NOTE: we cannot upgrade to the latest MINA since it looks like they have a bug in path resolution for remote directory creation.
Works OK on Windows though.

**Comment by artembilan** [comment]
Please, find the fix in the linked PR.

**Comment by ruedih** [comment]
Hey @artembilan ,

thanks for your super fast resolution of the issue. 🥇 :)
